### PR TITLE
Variabilize Helm charts versions and repositories of ArmoniK

### PR DESCRIPTION
**Pull request by YKharouni**
Variabilize the helm chart path repositories and the helm charts versions of ArmoniK.

fix : #890

### CODE REVIEW

File: infrastructure/quick-deploy/aws/storage/parameters.tfvars [hunk 1]
```diff
@@ -124,7 +124,8 @@ mongodb = {
   tag                = "6.0.1"
   node_selector      = { "grid/type" = "Operator" }
   image_pull_secrets = ""
-  persistent_volume  = null
+  #persistent_volume  = null
```
Comment (lzianekhodja) [inline infrastructure/quick-deploy/aws/storage/parameters.tfvars:127]: uncomment this line. By default, EFS is optional as persistent volume for MongoDB.

File: infrastructure/quick-deploy/aws/storage/parameters.tfvars [hunk 2]
```diff
@@ -124,7 +124,8 @@ mongodb = {
...
+  persistent_volume = { storage_provisioner = "efs.csi.aws.com", parameters = null, resources = { limits = null, requests = { storage = "5Gi" } } }
```
Comment (lzianekhodja) [inline infrastructure/quick-deploy/aws/storage/parameters.tfvars:128]: Delete this line.

File: infrastructure/quick-deploy/localhost/all/variables.tf [hunk 3]
```diff
@@ -212,8 +216,9 @@ variable "partition_metrics_exporter" {
 variable "fluent_bit" {
   description = "Fluent bit configuration"
   type = object({
-    image_name     = optional(string, "fluent/fluent-bit")
-    image_tag      = optional(string)
+    image_name = optional(string, "fluent/fluent-bit")
+    image_tag  = optional(string)
+    #pull_secrets   = optional(string, "")
```
Comment (lzianekhodja) [inline infrastructure/quick-deploy/localhost/all/variables.tf:221]: Delete this line.

File: infrastructure/quick-deploy/aws/eks/variables.tf [hunk 4]
```diff
@@ -75,6 +75,14 @@ variable "eks" {
       scale_down_delay_after_failure        = string
       scale_down_unneeded_time              = string
       skip_nodes_with_system_pods           = bool
+      version                               = optional(string, "9.24.0")
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/eks/variables.tf:78]: Do we allow optional for this deployment? I think this deployment should be kept compatible with versions of terraform before 1.3

File: infrastructure/quick-deploy/aws/all/variables.tf [hunk 5]
```diff
@@ -110,6 +110,14 @@ variable "eks" {
       scale_down_unneeded_time              = optional(string, "2m")
       skip_nodes_with_system_pods           = optional(bool, true)
       node_selector                         = optional(any, {})
+      version                               = optional(string, "9.24.0")
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/variables.tf:113]: Would it make sense to use a similar concept for the helm chart versions than for the docker image versions?

File: infrastructure/quick-deploy/aws/metrics-server/locals.tf [hunk 6]
```diff
@@ -11,5 +11,7 @@ locals {
     "--kubelet-use-node-status-port",
     "--metric-resolution=15s"
   ] : var.args
-  host_network = try(var.host_network, false)
+  host_network          = try(var.host_network, false)
+  helm_chart_repository = try(var.helm_chart_repository, "https://kubernetes-sigs.github.io/metrics-server/")
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/metrics-server/locals.tf:15]: This try cannot fail as the `var.helm_chart_repository` will always be valid (even if empty or null)

File: infrastructure/quick-deploy/aws/all/variables.tf (hunk 5 above)
Comment (YKharouni) [reply infrastructure/quick-deploy/aws/all/variables.tf:113]: We will do this in an other issue.

**Review by lzianekhodja — APPROVED** [review]

**Review by lzianekhodja — APPROVED** [review]

**Pull request merged**